app/utils/bundle-importer.js

```diff
@@ -460,7 +481,20 @@ YUI.add('bundle-importer', function(Y) {
               name = requiredModel.get('name') + '/' + size;
               break;
             case 2:
-              record.args[2] = requiredModel.id;
+              // Check whether this placement refers to a machine or a unit.
+              if (arg.indexOf('$addUnit') === 0) {
+                // We need to add a machine to specify colocation, and we also
+                // rely on _saveModelToRequires; _execute_addMachines is a
+                // helpful shortcut.
+                var machine = this._execute_addMachines({
+                  args: [{}],
+                  id: arg
+                }, function() {});
```

Comment:
I don't really like making the modification to `_execute_addMachines` because it 'special cases' that function by giving it two different exit paths compared to the rest of the `_execute_` methods. I propose having the next() method called passing it the `machine` then here you'd perform the remaining two lines (493, 494) in the supplied callback.